mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedParams.scala

```diff
@@ -357,4 +357,17 @@ private[ml] trait HasWeightCol extends Params {
   /** @group getParam */
```

Comment:
This file can not be modified directly. Please add `HasSolver` param to `SharedParamsCodeGen.scala`, and then run `build/sbt "mllib/runMain org.apache.spark.ml.param.shared.SharedParamsCodeGen"` to generate `sharedParams.scala`